rclcpp/test/rclcpp/test_timer_call_count.cpp

```diff
@@ -0,0 +1,119 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include <chrono>
+#include <string>
+#include <memory>
+
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp/exceptions.hpp"
+#include "rclcpp/node.hpp"
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp/executors.hpp"
+#include "rclcpp/executor.hpp"
+
+using namespace std::chrono_literals;
+
+class TestTimerCount: public ::testing::Test
+{
+protected:
+  static void SetUpTestCase()
+  {
+    rclcpp::init(0, nullptr);
+  }
+};
+
+constexpr int EXECUTION_COUNT = 5;
+constexpr double TIME_ELAPSED = 12.0;
+/*
+   Test timer wait mutex with multithreaded executor.
+   After 5 call
+ */
+
+TEST_F(TestTimerCount, timer_call_count_multi_threaded) {
```

Comment:
isn't this test conceptually similar to https://github.com/ros2/rclcpp/blob/master/rclcpp/test/rclcpp/executors/test_multi_threaded_executor.cpp?

Reply:
It is similar, but in that test it is checking that is waking up at least in its regular period minus some tolerance. In the test for this PR, I want to make sure that by the end of n seconds, the callbacks have been called n times the frequency desired.

Reply:
Sounds good